tests/e2e/test_log_and_fetch.py

```diff
@@ -217,3 +217,32 @@ def test_source_tracking(run, client, project_name):
         assert file_ref["mime_type"] == "application/patch"
         assert file_ref["size_bytes"] == len(value)
         assert file_ref["path"] is not None
+
+
+@pytest.mark.parametrize(
+    "inherit_configs",
+    [True, False],
+)
+def test_inherit_configs(api_token, client, run, project_name, inherit_configs):
+    # given
+    now = time.time()
+    data = {
+        "int-value": int(now),
+        "float-value": now,
+        "str-value": f"hello-{now}",
+    }
+    run.log_configs(data)
+    assert run.wait_for_processing(SYNC_TIMEOUT)
+
+    # when
+    with Run(
+        api_token=api_token, project=project_name, fork_run_id=run._run_id, fork_step=0, inherit_configs=inherit_configs
+    ) as run_2:
```

Comment:
I have a question about this test. If run_2 had also called `log_configs`, what would fetcher contain? The union of the keys with the last call overwriting the intersection? or only the keys from the second `log_config`s call?